riscv-rt/link.x.in

```diff
@@ -170,8 +170,8 @@ SECTIONS
     KEEP(*(.got .got.*));
   }
 
-  .eh_frame (INFO) : { KEEP(*(.eh_frame)) }
-  .eh_frame_hdr (INFO) : { *(.eh_frame_hdr) }
+  /* .eh_frame (INFO) : { KEEP(*(.eh_frame)) } */
+  /* .eh_frame_hdr (INFO) : { *(.eh_frame_hdr) } */
```

Comment:
```suggestion
  .eh_frame : { KEEP(*(.eh_frame)) } > REGION_TEXT
  .eh_frame_hdr (INFO) : { *(.eh_frame_hdr) } > REGION_TEXT
```

This is the suggestion from the linked issue, and also works to resolve the issue.

I think the main problem is that The `INFO` attribute clears the `SHF_ALLOC` bit in `sh_flags`: <https://lld.llvm.org/ELF/linker_script.html#output-section-type>

The `INFO` attribute is what looks to have been causing the error, since it was telling `rust-lld` to not allocate space for the `eh_frame` and `eh_frame_hdr` sections. Then, when referenced by another part of code, they were seen as being outside the allocated memory.

So, removing the `eh_frame` and `eh_frame_hdr` sections entirely, or removing the `INFO` section resolves the issue.

Completely removing the sections may be the better option, as it allows downstream users to define them when necessary, without forcing them to place them in the `REGION_TEXT` region.